### PR TITLE
base: replace Guava preconditions with conditionals

### PR DESCRIPTION
**Pull request by schildbach**
This has the added benefit of exception messages only being evaluated on demand.

### CODE REVIEW

File: core/src/main/java/org/bitcoinj/base/Bech32.java [hunk 1]
```diff
@@ -128,8 +126,10 @@ public static String encode(final Bech32Data bech32) {
 
     /** Encode a Bech32 string. */
     public static String encode(Encoding encoding, String hrp, final byte[] values) {
-        checkArgument(hrp.length() >= 1, "Human-readable part is too short");
-        checkArgument(hrp.length() <= 83, "Human-readable part is too long");
+        if (hrp.length() < 1)
+            throw new IllegalArgumentException("human-readable part is too short: " + hrp.length());
+        if (hrp.length() > 83)
+            throw new IllegalArgumentException("human-readable part is too long: " + hrp.length());
```
Comment (msgilligan) [inline core/src/main/java/org/bitcoinj/base/Bech32.java:132]: I think we could make an exception (or new rule) to allow `if (cons) throw exception` on 1-line for argument checking.

File: core/src/main/java/org/bitcoinj/base/LegacyAddress.java [hunk 2]
```diff
@@ -66,11 +64,15 @@ public class LegacyAddress implements Address {
      *            20-byte hash of pubkey or script
      */
     private LegacyAddress(Network network, boolean p2sh, byte[] hash160) throws AddressFormatException {
-        this.network = normalizeNetwork(checkNotNull(network));
-        this.bytes = checkNotNull(hash160);
+        if (network == null)
+            throw new NullPointerException();
```
Comment (msgilligan) [inline core/src/main/java/org/bitcoinj/base/LegacyAddress.java:68]: This could be `Objects.requireNonNull(network)`

Comment (msgilligan) [reply core/src/main/java/org/bitcoinj/base/LegacyAddress.java:68]: Or even:
```
this.network = normalizeNetwork(Objects.requireNonNull(network));
```

We could also us a static import for `Objects.requireNonNull`.

**Review by msgilligan — CHANGES_REQUESTED** [review]
I definitely think we should use `Objects.requireNonNull` where possible.

I also prefer the brevity of `checkState` and think we should consider creating an equivalent static utility method.

Note that our utility method could take a `Supplier<String>` argument, so invocation would look like:
```
checkArgument(hrp.length() >= 1, () -> "human-readable part is too short: " + hrp.length());
```

**Comment by schildbach** [comment]
This PR should be reviewed and merged first: https://github.com/bitcoinj/bitcoinj/pull/2819

I will then rebase this one and address your comments.

File: core/src/main/java/org/bitcoinj/base/LegacyAddress.java (hunk 2 above)
Comment (schildbach) [reply core/src/main/java/org/bitcoinj/base/LegacyAddress.java]: This has been resolved via #2819.

**Comment by schildbach** [comment]
Rebased, and added Preconditions utility methods.

Ready for another review.

File: core/src/main/java/org/bitcoinj/base/Bech32.java (hunk 1 above)
Comment (schildbach) [reply core/src/main/java/org/bitcoinj/base/Bech32.java]: In the context of this PR, this is no longer necessary.

**Review by msgilligan — APPROVED** [review]
Looks great!

**Comment by schildbach** [comment]
Merged.

**Pull request closed without merging**